### DISCUSSION TITLE
MSL - authenticate attribute parsing

### DISCUSSION THREAD

**Discussion by J0hnnyB0y** [ImageMagick/ImageMagick#2779] [Development]
Hi.

I started to have some interest in the MSL support of ImageMagick so I started to read the code. Most of it is quite straightforward but I was wondering about the "authenticate" attribute support:

https://github.com/ImageMagick/ImageMagick/blob/master/coders/msl.c#L8134

if (LocaleCompare(keyword,"authenticate") == 0)
        {
          (void) CloneString(&image_info->density,value);
          break;
        }

Why is the defined authenticate value assigned to the density attribute? Isn't this already handled in https://github.com/ImageMagick/ImageMagick/blob/master/coders/msl.c#L8177, which is the density attribute?^^

Shouldn't authenticate handle authenticate eg a password as used in coders/pdf.c or MagickCore/delegate.c?

cheers

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Reply by J0hnnyB0y** [reply]
Thanks for the fast respond :) 
I checked the new code and I think there is a mistake:
      if (LocaleCompare(keyword,"authenticate") == 0)
        {
          (void) SetImageOption(image_info,keyword,keyword);
          break;
        }

Shouldn't be the value passed ?

**Reply by dlemstra** [reply]
Good catch tnx 👍
